Xamarin.Forms.Platform.Tizen/Platform.cs

```diff
@@ -562,6 +564,97 @@ void ActionSheetSignalNameHandler(Page sender, ActionSheetArguments arguments)
 			alert.Dismissed += (s, e) => _alerts.Remove(alert);
 		}
 
+		void OnPromptRequested(Page sender, PromptArguments args)
+		{
+			// Verify that the page making the request is child of this platform
+			if (!PageIsChildOfPlatform(sender))
+				return;
+
+			var prompt = Native.Dialog.CreateDialog(Forms.NativeParent, (args.Accept != null));
+			prompt.Title = args.Title;
+
+			var entry = new Entry
+			{
+				MinimumWidthRequest = 200,
+				HorizontalOptions = LayoutOptions.FillAndExpand,
+				BackgroundColor = Color.FromRgb(250, 250, 250),
+				TextColor = Color.Black,
+				Keyboard = args.Keyboard,
+			};
+
+			if (!string.IsNullOrEmpty(args.Placeholder))
+			{
+				entry.Placeholder = args.Placeholder;
+			}
+			if (args.MaxLength > 0)
+			{
+				entry.MaxLength = args.MaxLength;
+			}
+
+			var layout = new StackLayout
+			{
+				Spacing = 10,
+				Children =
+				{
+					new Label
+					{
+						LineBreakMode = LineBreakMode.CharacterWrap,
+						TextColor = Device.Idiom == TargetIdiom.Watch ? Color.White : Color.Accent,
+						Text = args.Message,
+						HorizontalOptions = LayoutOptions.FillAndExpand,
+						HorizontalTextAlignment = TextAlignment.Center,
+						FontSize = Device.GetNamedSize(NamedSize.Subtitle, typeof(Label)),
+					},
+					entry,
+				}
+			};
+
+			layout.Parent = sender;
+			var layoutrenderer = Platform.GetOrCreateRenderer(layout);
+
+			var request = layout.Measure(Device.Idiom == TargetIdiom.Watch ? Page.Width * 0.7 : Page.Width, Page.Height);
+			(layoutrenderer as LayoutRenderer).RegisterOnLayoutUpdated();
+			layoutrenderer.NativeView.MinimumHeight = Forms.ConvertToScaledPixel(request.Request.Height);
+			layoutrenderer.NativeView.MinimumWidth = Forms.ConvertToScaledPixel(request.Request.Width);
+
+			prompt.Content = layoutrenderer.NativeView;
+
+			var cancel = new EButton(prompt) { Text = args.Cancel };
+			prompt.NegativeButton = cancel;
+			cancel.Clicked += (s, evt) =>
+			{
+				args.SetResult(null);
+				prompt.Dismiss();
+			};
+
+			if (args.Accept != null)
+			{
+				var ok = new EButton(prompt) { Text = args.Accept };
+				prompt.NeutralButton = ok;
+				ok.Clicked += (s, evt) =>
+				{
+					args.SetResult(entry.Text);
+					prompt.Dismiss();
+				};
+			}
+
+			entry.Completed += (s, e) =>
+			{
+				args.SetResult(entry.Text);
+				prompt.Dismiss();
+			};
+
+			prompt.BackButtonPressed += (s, evt) =>
+			{
+				prompt.Dismiss();
```

Comment:
Should the result also be set here because otherwise the tcs never completes in this case?

```suggestion
				args.SetResult(null);
				prompt.Dismiss();
```

Reply:
@campersau Yes, you are right!